src/sentry/notifications/platform/target.py

```diff
@@ -66,4 +74,26 @@ def prepare_targets(targets: list[NotificationTarget]) -> None:
     `organization_integration` fields by making RPC/DB calls.
     """
     for target in targets:
+        if target.provider_key in INTEGRATION_PROVIDER_KEYS:
+            assert isinstance(
+                target, IntegrationNotificationTarget
+            ), "Target for integration providers must be an IntegrationNotificationTarget"
+            prepare_integration_target(target)
+
         object.__setattr__(target, "is_prepared", True)
+
+
+def prepare_integration_target(target: IntegrationNotificationTarget) -> None:
+    """
+    This method is used to prepare the integration target by populating the `integration` and
+    `organization_integration` fields.
+    """
+    integration = integration_service.get_integration(integration_id=target.integration_id)
+    organization_integration = integration_service.get_organization_integration(
+        integration_id=target.integration_id,
+        organization_id=target.organization_id,
+    )
+    if integration is not None:
+        object.__setattr__(target, "integration", integration)
+    if organization_integration is not None:
+        object.__setattr__(target, "organization_integration", organization_integration)
```

Comment:
Could we just return an installation of the desired type instead of appending it to the notification target?

Another option: we could compose both the target and installation in some higher-order class which would be a bit cleaner imo. It also gives us a place to add helper methods + turn this into a factory method.